### PR TITLE
Output tasks using GitHub task lists

### PR DESCRIPTION
**Pull request by RealOrangeOne**
This allows us to use more of GitHub's project features

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/about-task-lists

Fixes https://github.com/srobo/recurring-tasks/issues/73

### CODE REVIEW

**Review by WillB97 — APPROVED** [review]

**Comment by PeterJCLaw** [comment]
Nice, do you have a demo import of this somewhere you've tested with?

**Comment by RealOrangeOne** [comment]
Untested yet, mostly opening as an RFC. Will test soon.

File: scripts/import_backends.py [hunk 1]
```diff
@@ -227,7 +227,7 @@ def description_text(self, ticket: Ticket) -> str:
 
         if ticket.dependencies:
             text += self._section("Dependencies", "\n".join(
-                f" * #{dep} {self.title(dep)}".rstrip()
+                f" - [ ] #{dep} {self.title(dep)}".rstrip()
```
Comment (PeterJCLaw) [inline scripts/import_backends.py:230]: If I recall GitHub includes the titles too if we do this -- should we drop our explicit inclusion of the title?

**Comment by PeterJCLaw** [comment]
> Nice, do you have a demo import of this somewhere you've tested with? (only needs to be a couple of tasks, no need for a whole tree)

I've imported a test tree into https://github.com/PeterJCLaw/test/issues/817

**Comment by PeterJCLaw** [comment]
I've extended this with deeper task-list stuff. Example imports:
- https://github.com/srobo/test/issues/9
- https://github.com/srobo/test/issues/6
- https://github.com/srobo/test/issues/3

**Review by prophile — APPROVED** [review]

**Review by RealOrangeOne — COMMENTED** [review]
LGTM :heavy_check_mark:

**Pull request merged**